polling_stations/apps/file_uploads/templates/file_uploads/index.html

```diff
@@ -0,0 +1,178 @@
+  var error = '<h2>Sorry, something has gone wrong</h2> <p>If the problem continues, ';
+  error += '<a href="mailto:pollingstations@democracyclub.org.uk">contact us</a>.</p>';
+  document.getElementById("error").innerHTML = error;
+  document.getElementById("error").hidden = false;
+  document.getElementById("submit").disabled = false;
+};
+
+var serializeFile = function(file) {
+  if (file == null) return null;
+  return {
+    name: file.name,
+    size: file.size,
+    type: file.type,
+  };
+};
+
+var getPresignedPostData = function() {
+  return new Promise(function(resolve, reject) {
+    var xhr = new XMLHttpRequest();
+    var url = '{% url "file_upload_index" %}';
+    xhr.open("POST", url, true);
+    xhr.setRequestHeader("Content-Type", "application/json");
+    xhr.setRequestHeader("X-CSRFToken", getCookie('csrftoken'));
+    var payload = {
+      files: [
+        serializeFile(document.getElementById('file0').files[0]),
+        serializeFile(document.getElementById('file1').files[0])
+      ].filter(Boolean),
+      council_id: document.getElementById('council_id').value
+    };
+    xhr.send(JSON.stringify(payload));
+    xhr.onload = function() {
+      if (this.status === 200) {
+        resolve(JSON.parse(this.responseText));
+       } else {
+        reject(this.responseText);
+       }
+    };
+  });
+};
+
+var uploadFileToS3 = function(presignedPostData, file, progressBar) {
+  return new Promise(function(resolve, reject) {
+    var formData = new FormData();
+    Object.keys(presignedPostData.fields).forEach(function(key) {
+      formData.append(key, presignedPostData.fields[key]);
+    });
+
+    formData.append("file", document.getElementById(file).files[0]);
+
+    var xhr = new XMLHttpRequest();
+    xhr.open("POST", presignedPostData.url, true);
+    xhr.upload.addEventListener("progress", function(event) {
+      document.getElementById(progressBar).value = (event.loaded / event.total) * 100;
+    }, false);
+    xhr.send(formData);
+    xhr.onload = function() {
+      if (this.status === 204) {
+        resolve(this.responseText);
+      } else {
+        reject(this.responseText);
+      }
+    };
+  });
+};
+
+document.addEventListener('DOMContentLoaded', function() {
+
+  document.getElementById('file_upload_form').addEventListener("submit", function (e) {
+    e.preventDefault();
+    document.getElementById("error").innerHTML = "";
+    document.getElementById("error").hidden = true;
+    document.getElementById("submit").disabled = true;
+
+    getPresignedPostData().then(
+      function(data) {
+        var uploads = [];
+        for (var i=0; i<data.files.length; i++) {
+          uploads.push(
+            uploadFileToS3(data.files[i], 'file'.concat(i), 'progressBar'.concat(i))
+          );
+        }
+        Promise.all(uploads).then(
+          function(data) {
+            console.log('done!');
+            console.log(data);
+            document.getElementById("submit").disabled = false;
```

Comment:
TODO: implement some kind of success behaviour.. not sure what that looks like yet